myapp/views.py

```diff
@@ -88,28 +88,20 @@ def watchlist_view(request):
 
 @require_http_methods(['POST'])
 @login_required(login_url='login')
-def watchlist_add_view(request, symbol):
+def watchlist_edit_view(request, symbol, operation):
     profile = Profile.objects.get(user=request.user)
-    stock = Stock.objects.filter(symbol=symbol)
+    stock = Stock.objects.filter(symbol=symbol)[:1]
     if not stock.exists():
-        raise Http404("Stock does not exist")
-    else:
-        Stock.add_to_watchlist(profile, symbol)
-        next = request.POST.get('next', '/')
-        return redirect(next)
-
-
-@require_http_methods(['POST'])
-@login_required(login_url='login')
-def watchlist_remove_view(request, symbol):
-    profile = Profile.objects.get(user=request.user)
-    stock = Stock.objects.filter(symbol=symbol)
-    if not stock.exists():
-        raise Http404("Stock does not exist")
+        response = render(request, "exception.html", {'error_message': 'Stock symbol not found', 'status_code': 404})
+        response.status_code = 404
+        return response
     else:
-        Stock.remove_from_watchlist(profile, symbol)
-        next = request.POST.get('next', '/')
-        return redirect(next)
+        if operation == "wremove":
+            Stock.remove_from_watchlist(profile, symbol)
+        if operation == "wadd":
+            Stock.add_to_watchlist(profile, symbol)
+        nextpage = request.POST.get('next', '/')
+        return redirect(nextpage)               # NOTE: last two lines no longer needed, remove?
```

Comment:
I'm not sure why "next" is needed and where is assigned to the value

Reply:
It is no longer needed, lines 103 and 104 can be safely deleted

Reply:
It is no longer needed, we can return some other response

Reply:
you can change to `return HttpResponse("OK")`  (redirecting cause to create redundant request)